### PR TITLE
chore: add decimal corrections for payment-helper

### PR DESCRIPTION
**Pull request by sujithsomraaj**

### CODE REVIEW

File: src/payments/PaymentHelper.sol [hunk 1]
```diff
@@ -50,7 +50,8 @@ contract PaymentHelper is IPaymentHelper {
     //////////////////////////////////////////////////////////////
 
     uint256 private constant PROOF_LENGTH = 160;
-    uint8 private constant SUPPORTED_FEED_PRECISION = 8;
+    uint8 private constant MIN_FEED_PRECISION = 8;
+    uint8 private constant MAX_FEED_PRECISION = 8;
```
Comment (0xTimepunk) [inline src/payments/PaymentHelper.sol:54]: isn't this 18?

File: src/payments/PaymentHelper.sol [hunk 2]
```diff
@@ -999,7 +1005,11 @@ contract PaymentHelper is IPaymentHelper {
             ) {
                 if (value <= 0) revert Error.CHAINLINK_MALFUNCTION();
                 if (updatedAt == 0) revert Error.CHAINLINK_INCOMPLETE_ROUND();
-                return uint256(value);
+
+                uint256 oraclePrecision = _getOracleDecimals(AggregatorV3Interface(oracleAddr));
+
+                if (oraclePrecision == MIN_FEED_PRECISION) return uint256(value);
+                else return uint256(value) / (10 ** oraclePrecision - MIN_FEED_PRECISION);
```
Comment (0xTimepunk) [inline src/payments/PaymentHelper.sol:1012]: Validate this calculation with eOracle. Also shouldn't we do 10 **(oraclePrecision - MIN_FEED_PRECISION) ?

**Review by 0xTimepunk — COMMENTED** [review]
Requires re-review

File: src/payments/PaymentHelper.sol [hunk 3]
```diff
@@ -999,7 +1005,11 @@ contract PaymentHelper is IPaymentHelper {
             ) {
                 if (value <= 0) revert Error.CHAINLINK_MALFUNCTION();
                 if (updatedAt == 0) revert Error.CHAINLINK_INCOMPLETE_ROUND();
-                return uint256(value);
+
+                uint256 oraclePrecision = _getOracleDecimals(AggregatorV3Interface(oracleAddr));
```
Comment (gpersoon) [inline src/payments/PaymentHelper.sol:1009]: This piece of code retrieves the decimals every time from the oracle
If you want to save gas, this value could be combined with the oracle address and stored in one slot.
This does make the code more complex though

**Pull request merged**